LoadingScreen/EventHandlers.cs

```diff
@@ -1,105 +1,101 @@
-﻿using System;
+using System;
+using System.Collections.Generic;
 using System.Linq;
-using ServerMod2.API;
 using Smod2.API;
 using Smod2.EventHandlers;
 using Smod2.Events;
-using UnityEngine;
-using Random = System.Random;
 
 namespace LoadingScreen
 {
     public class EventHandlers : IEventHandlerPlayerJoin, IEventHandlerWaitingForPlayers, IEventHandlerRoundStart, IEventHandlerFixedUpdate
     {
         private readonly LoadingScreen plugin;
 
-        private Smod2.API.Door[] doors;
-        
-        private Random rnd = new Random();
+        private Door[] doors = { };
 
-        private bool preRound = false;
+        private readonly Random rnd = new Random();
+
+        private DateTime? nextRefreshTime;
+        private bool preRound;
 
         public EventHandlers(LoadingScreen plugin)
         {
             this.plugin = plugin;
         }
 
-        public void OnWaitingForPlayers(WaitingForPlayersEvent ev)
+        public Door GetRndDoor()
         {
-            plugin.RefreshConfig();
-            if (plugin.doors.Contains("*"))
+            return doors.Any() ? doors[rnd.Next(0, doors.Length)] : null;
+        }
+
+        public static void TeleportToDoor(Player player, Door door)
+        {
+            if (door != null && player != null && player.TeamRole.Team == Team.NONE)
             {
-                /*Door escape = (Door) ev.Server.Map.GetDoors().First(x => x.Name.Equals("ESCAPE")).GetComponent();
-                Quaternion rot = escape.localRot;
-                */
-                doors = ev.Server.Map.GetDoors()
-                    /*.Where(x =>
-                {
-                    if (x.Name.Equals("SURFACE_GATE"))
-                        return false;
-                    Quaternion rotation = ((Door) x.GetComponent()).localRot;
-                    Quaternion rRotation = new Quaternion(-rotation.x,-rotation.y,rotation.z,rotation.w);
-                    bool result = rot == rotation || rot == rRotation;
-                    plugin.Info($"{x.Name} : {result}");
-                    return result;
-                })
-                */.ToArray();
+                player.Teleport(door.Position + Vector.Up);
```

Comment:
i'll thanks you cause this are really good changes, i'll only keep the teleport add vector to the Vector.One cause the vector up will let the player be stuck in the door if the door is 90° rotated than player POV